kv/kvserver: TestStoreRangeSplitWithConcurrentWrites failed

kv/kvserver.TestStoreRangeSplitWithConcurrentWrites [failed](https://tanzanite.cluster.engflow.com/invocations/default/ef5cc17d-912c-496a-a960-013c56e5d282?testReportRun=1&testReportShard=39&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.3 @ [eaa38b5fe50c68d6de15a17b984d937c8af6f94d](https://github.com/cockroachdb/cockroach/commits/eaa38b5fe50c68d6de15a17b984d937c8af6f94d):


```
=== RUN   TestStoreRangeSplitWithConcurrentWrites
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestStoreRangeSplitWithConcurrentWrites3260229422
    test_log_scope.go:76: use -show-logs to present logs inline
    client_split_test.go:1077: -- test log scope end --
test logs left over in: outputs.zip/logTestStoreRangeSplitWithConcurrentWrites3260229422
--- FAIL: TestStoreRangeSplitWithConcurrentWrites (57.37s)
=== RUN   TestStoreRangeSplitWithConcurrentWrites/estimates=true
    --- FAIL: TestStoreRangeSplitWithConcurrentWrites/estimates=true (29.07s)
=== RUN   TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true
        --- FAIL: TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true (14.66s)
=== RUN   TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false
            --- FAIL: TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false (7.29s)
=== RUN   TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false/maxBytes=false
    client_split_test.go:1066: 
        	Error Trace:	pkg/kv/kvserver/client_split_test.go:1066
        	            				pkg/testutils/subtest.go:23
        	Error:      	"0" is not greater than "0"
        	Test:       	TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false/maxBytes=false
                --- FAIL: TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false/maxBytes=false (3.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=39</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitWithConcurrentWrites.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46968

kv/kvserver.TestStoreRangeSplitWithConcurrentWrites [failed](https://tanzanite.cluster.engflow.com/invocations/default/b79ca037-d827-441a-92f4-669499d9947b?testReportRun=2&testReportShard=44&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.1 @ [ec1d001c2e0781b5ba46d4556812dc3180d0c86d](https://github.com/cockroachdb/cockroach/commits/ec1d001c2e0781b5ba46d4556812dc3180d0c86d):


```
=== RUN   TestStoreRangeSplitWithConcurrentWrites
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestStoreRangeSplitWithConcurrentWrites1988708744
    test_log_scope.go:76: use -show-logs to present logs inline
    client_split_test.go:1077: -- test log scope end --
test logs left over in: outputs.zip/logTestStoreRangeSplitWithConcurrentWrites1988708744
--- FAIL: TestStoreRangeSplitWithConcurrentWrites (37.16s)
=== RUN   TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true
        --- FAIL: TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true (8.81s)
=== RUN   TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false
            --- FAIL: TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false (4.44s)
=== RUN   TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false/maxBytes=false
    client_split_test.go:1066: 
        	Error Trace:	pkg/kv/kvserver/client_split_test.go:1066
        	            				pkg/testutils/subtest.go:23
        	Error:      	"0" is not greater than "0"
        	Test:       	TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false/maxBytes=false
                --- FAIL: TestStoreRangeSplitWithConcurrentWrites/estimates=true/recompute=true/maxCount=false/maxBytes=false (2.20s)
=== RUN   TestStoreRangeSplitWithConcurrentWrites/estimates=true
    --- FAIL: TestStoreRangeSplitWithConcurrentWrites/estimates=true (18.28s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=44</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitWithConcurrentWrites.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/client_split_test.go
+++ b/pkg/kv/kvserver/client_split_test.go
@@ -957,9 +957,10 @@ func TestStoreRangeSplitWithConcurrentWrites(t *testing.T) {
 					s := serverutils.StartServerOnly(t, base.TestServerArgs{
 						Knobs: base.TestingKnobs{
 							Store: &kvserver.StoreTestingKnobs{
-								DisableMergeQueue:    true,
-								DisableSplitQueue:    true,
-								TestingRequestFilter: filter,
+								DisableMergeQueue:       true,
+								DisableSplitQueue:       true,
+								DisableConsistencyQueue: true,
+								TestingRequestFilter:    filter,
 							},
 						},
 						Settings: settings,
@@ -1061,8 +1062,7 @@ func TestStoreRangeSplitWithConcurrentWrites(t *testing.T) {
 						assertRecomputedStats(t, "LHS2 after second split", snap, lhs2Repl.Desc(), lhs2Stats, s.Clock().PhysicalNow())
 						assertRecomputedStats(t, "RHS1 after second split", snap, rhsRepl.Desc(), rhs1Stats, s.Clock().PhysicalNow())
 						assertRecomputedStats(t, "RHS2 after second split", snap, rhs2Repl.Desc(), rhs2Stats, s.Clock().PhysicalNow())
-					}
-					if expectContainsEstimates {
+					} else {
 						require.Greater(t, lhs1Stats.ContainsEstimates, int64(0))
 						require.Greater(t, lhs2Stats.ContainsEstimates, int64(0))
 						// The range corresponding to rhs1Stats is empty, so the split of the

